src/convert3Dto4D.m

```diff
@@ -0,0 +1,154 @@
+
+  tic;
+
+  for iSeq = 1:size(optSource.sequenceList, 1)
+
+    % Skip 'non' folders
+    if length(optSource.sequenceList{iSeq}) > 2
+
+      % Check if sequence to ignore or not
+      if contains(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)
+
+        warning('\nIGNORING SEQUENCE: %s\n', string(optSource.sequenceList(iSeq)));
+
+      else
+
+        fprintf('\n\nCONVERTING SEQUENCE: %s \n', char(optSource.sequenceList(iSeq)));
+
+        % Set whether to remove dummies or not
+
+        nbDummies = 0;
+
+        if contains(optSource.sequenceList(iSeq), optSource.sequenceRmDummies)
+
+          nbDummies = optSource.nbDummies;
+
+          fprintf('\n\nREMOVING %s DUMMIES\n\n', num2str(nbDummies));
+
+        end
+
+        % Get sequence folder path
+        sequencePath = fullfile(optSource.sourceDir, optSource.sequenceList{iSeq});
+
+        % Retrieve volume files info
+        [volumesList, outputNameImage] = parseFiles('nii', sequencePath, nbDummies);
+
+        % Set output name, it takes the file name of the 1st volume of the 4D file and add subfix
+        outputNameImage = strrep(outputNameImage, '.nii', '_4D.nii');
+
+        % Retrieve sidecar json files info
+        [jsonList, outputNameJson] = parseFiles('json', sequencePath, nbDummies);
+
+        jsonFile = spm_jsonread(jsonList{1});
+
+        % % % % % % LIEGE  SPECIFIC % % % % % % %
+        RT = jsonFile.acqpar.RepetitionTime / 1000;
+        % % % % % % % % % % % % % % % % % % % % %
+
+        % Set and run spm batch, input all the volumes minus the dummies if > 0
+        matlabbatch = [];
+        matlabbatch = setBatch3Dto4D(matlabbatch, ...
+                                     volumesList(nbDummies + 1:end, :), ...
+                                     RT, ...
+                                     outputNameImage, ...
+                                     optSource.dataType);
+
+        spm_jobman('run', matlabbatch);
+
+        if optSource.zip
+
+          % Zip and delete the and the new 4D file
+          fprintf(1, 'ZIP AND DELETE THE NEW 4D BRAIN \n\n');
+
+          gzip([sequencePath filesep outputNameImage]);
+
+          delete([sequencePath filesep outputNameImage]);
+
+        end
+
+        % Save one sidecar json file, it takes the file name of the 1st volume of the 4D file and
+        % add subfix
+        if ~isempty(jsonList)
+
+          copyfile(jsonList{1}, [sequencePath filesep strrep(outputNameJson, '.json', '_4D.json')]);
+
+        end
+
+        % Delete all the single volumes .nii and .json files
+        fprintf(1, 'EXTERMINATE SINGLE VOLUMES FILES \n\n');
+
+        for iDel = 1:length(volumesList)
+
+          delete(volumesList{iDel});
+          delete(jsonList{iDel});
+
+        end
+
+      end
+
+    end
+
+  end
+
+  toc;
+
+end
+
+function [fileList, outputName] = parseFiles(fileExtention, sequencePath, nbDummies)
+
+  fileList = spm_select('list', sequencePath, fileExtention);
```

Comment:
I think that using the full path option here might save you some trouble.

it might even make part (or the whole subfunction) unnecessary.

```
fileList = spm_select('FPlist', sequencePath, fileExtention);
```

Reply:
Will check, opening an issue about this